src/vivarium_public_health/metrics/mortality.py

```diff
@@ -78,40 +81,84 @@ def setup(self, builder: Builder) -> None:
         self.causes_of_death = ["other_causes"] + [
             cause.state_id for cause in self._cause_components if cause.has_excess_mortality
         ]
+        if not self.config.aggregate:
+            self._register_cause_specific_observations(builder)
+        else:
+            self._register_aggregate_observations(builder)
 
+    ###################
+    # Private methods #
+    ###################
+
+    def _register_cause_specific_observations(self, builder: Builder) -> None:
         for cause_of_death in self.causes_of_death:
-            builder.results.register_observation(
+            self._register_observation(
+                builder=builder,
                 name=f"death_due_to_{cause_of_death}",
                 pop_filter=f'alive == "dead" and cause_of_death == "{cause_of_death}"',
-                aggregator=self.count_cause_specific_deaths,
+                aggregator=self.count_deaths,
                 requires_columns=["alive", "cause_of_death", "exit_time"],
-                additional_stratifications=self.config.include,
-                excluded_stratifications=self.config.exclude,
-                when="collect_metrics",
             )
-            builder.results.register_observation(
+            self._register_observation(
+                builder=builder,
                 name=f"ylls_due_to_{cause_of_death}",
                 pop_filter=f'alive == "dead" and cause_of_death == "{cause_of_death}"',
-                aggregator=self.calculate_cause_specific_ylls,
+                aggregator=self.calculate_ylls,
                 requires_columns=[
                     "alive",
                     "cause_of_death",
                     "exit_time",
                     "years_of_life_lost",
                 ],
-                additional_stratifications=self.config.include,
-                excluded_stratifications=self.config.exclude,
-                when="collect_metrics",
             )
 
+    def _register_aggregate_observations(self, builder: Builder) -> None:
+        self._register_observation(
```

Comment:
I was envisioning a different way to reduce code duplication. Let's talk about it after standup